docs/reserved-keywords.adoc

```diff
@@ -0,0 +1,278 @@
+= PartiQL Reserved Keywords
+
+Reserved keywords are part of PartiQL Language that are used by PartiQL to parse and understand PartiQL statements.
+PartiQL Language Grammar Reserved Keywords cannot be used in PartiQL statements as-is.
+For example, the following query results in a _Syntax Error_ because `FROM` is a Reserved Keyword in PartiQL:
+
+.Example 1. Invalid PartiQL Query with a Reserved Keyword
+[source, sql]
+----
+SELECT t.FROM FROM <<{'ID': 1, 'FROM': 'NA'}, {'ID': 2, 'FROM': 'SA'}>> AS t;
+----
+
+In order to use Reserved Keywords in your PartiQL statements, use Quoted Identifiers as Example 2 shows:
+
+.Example 2. Using Quoted Identifier as a Reserve Keyword in a PartiQL Statement.
+[source, sql]
+----
+SELECT t."FROM" FROM <<{'ID': 1, 'FROM': 'NA'}, {'ID': 2, 'FROM': 'SA'}>> AS t;
```

Comment:
Probably want to also call out that the reserved keyword variable reference won't be case sensitive since you use double quotes.

Reply:
Makes sense; I'll point that out.

Also as they are going to be case-sensitive after putting between double-quotes; I'll clarify that too:

```
PartiQL> SELECT DISTINCT t."FROM" FROM <<{'ID': 1, 'from': 'NA'}, {'ID': 2, 'from': 'SA'}>> AS t;
==='
<<
  {}
>>
---
OK!

PartiQL> SELECT DISTINCT t."from" FROM <<{'ID': 1, 'FROM': 'NA'}, {'ID': 2, 'FROM': 'SA'}>> AS t;
==='
<<
  {}
>>
---
OK!

PartiQL> SELECT DISTINCT t."from" FROM <<{'ID': 1, 'FROM': 'NA'}, {'ID': 2, 'from': 'SA'}>> AS t;
==='
<<
  {}
>>
---
OK!
```